api/src/main/java/org/openmrs/module/pihcore/setup/MetadataSetupTask.java

```diff
@@ -0,0 +1,85 @@
+package org.openmrs.module.pihcore.setup;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.metadatadeploy.api.MetadataDeployService;
+import org.openmrs.module.pihcore.config.Config;
+import org.openmrs.module.pihcore.config.ConfigDescriptor;
+import org.openmrs.module.pihcore.deploy.bundle.haiti.HaitiMetadataToInstallAfterConceptsBundle;
+import org.openmrs.module.pihcore.deploy.bundle.liberia.LiberiaMetadataToInstallAfterConceptsBundle;
+import org.openmrs.module.pihcore.deploy.bundle.mexico.MexicoMetadataToInstallAfterConceptsBundle;
+import org.openmrs.module.pihcore.deploy.bundle.sierraLeone.SierraLeoneMetadataToInstallAfterConceptsBundle;
+
+public class MetadataSetupTask implements Runnable {
+
+    protected Log log = LogFactory.getLog(getClass());
+
+    private Config config;
+
+    // hack so that we can disable this during testing, because we are not currently installing MDS packages as part of test
+    private Boolean disableInstallMetadataBundlesThatDependOnMDSPackages = false;
+
+    public MetadataSetupTask(Config config, Boolean disableInstallMetadataBundlesThatDependOnMDSPackages) {
+        this.config = config;
+        this.disableInstallMetadataBundlesThatDependOnMDSPackages = disableInstallMetadataBundlesThatDependOnMDSPackages;
+    }
+
+    @Override
+    public void run()  {
+
+        try {
+            MetadataSharingSetup.installMetadataSharingPackages();
+            log.info("Metadata sharing packages loaded");
+        }
+        catch (Exception e) {
+            log.error("Aborting Metadata Setup Task: error installing Metadata Sharing Packages", e);
+            throw e;
+        }
+
+        if (!this.disableInstallMetadataBundlesThatDependOnMDSPackages) {
+            try {
+                installMetadataBundlesThatDependOnMDSPackages(config);
+                log.info("Metadata bundles dependant on MDS packages loaded");
+            }
+            catch (Exception e) {
+                log.error("Aborting Metadata Setup Task: error installing dependant Metadata bundles", e);
+                throw e;
+            }
+        }
+
+        try {
+            DrugListSetup.installDrugList();
+            log.info("Drug list loaded");
+        }
+        catch (Exception e) {
+            log.error("Aborting Metadata Setup Task: error installing drug list", e);
+            throw e;
+        }
+
+        try {
+            HtmlFormSetup.loadHtmlForms(); // this must happen *after* MDS packages are installed, so that forms defined in code/github take precedent
+            log.info("HTML forms loaded");
+        }
+        catch (Exception e) {
+            log.error("Aborting Metadata Setup Task: error installing HTML forms", e);
+            throw e;
+        }
+    }
+
```

Comment:
LGTM, but maybe just rename this PR / ticket to reflect it is more than just Concepts in Background